packages/server/src/oauth/utils.ts

```diff
@@ -501,15 +501,12 @@ export async function setLoginScope(login: Login, scope: string): Promise<Login>
 }
 
 export async function getAuthTokens(
+  user: User | ClientApplication,
   login: Login,
   profile: Reference<ProfileResource>,
   refreshLifetime?: string
 ): Promise<TokenResult> {
   const clientId = login.client && resolveId(login.client);
-  const userId = resolveId(login.user);
```

Comment:
Should we assert that the `user` and `login.user` match here before we continue?

Reply:
`getAuthTokens` is only called in 2 places, and that is guaranteed to be true.

I suppose we could do that for defensive futureproofing, but I consider that low pri in this case.

Reply:
There are 2 call sites:
1. `register.ts` - on registering a new user, relatively rare event
2. `token.ts` - when sending a token response, relatively common event

In `token.ts`, we immediately read `user` and `profile` right before this.  If we're ok with double reading in the `register.ts` case, we could move that into `getAuthTokens`, then you don't need to worry about the assumptions.

Reply:
Hm, k, I tried that, and things get messy, because you need the `user` and `membership` outside of `getAuthTokens` as well, so that would be a bigger refactor.

Bottom line, I guess my preference would be to leave as-is.

Reply:
Will do